src/semgrep_agent/main.py

```diff
@@ -257,9 +257,7 @@ def main(
 
     if json_output:
         # Output all new findings as json
-        json_contents = [
-            f.to_dict(omit=constants.PRIVACY_SENSITIVE_FIELDS) for f in new_findings
-        ]
+        json_contents = [f.to_dict(omit=set()) for f in new_findings]
```

Comment:
Do you need the omit=set() here? I would guess it works the same without that

Reply:
or is it because by default to_dict was rewritten to use PRIVACY_SENSITIVE_FIELDS? Probably worth adding a comment explaining that

Reply:
yeah, that's what I did, but I'm not sure; if we had to add a comment to all its usages, then we're better off making it a required keyword argument with no default

do you think this confused you only because you were familiar with the old behavior, or would it be confusing to anyone reading this code for the first time?

Reply:
I think the only reason it _didn't_ confuse me was because the new to_dict() definition was right above it in the git diff. What do you think of making the required argument be a boolean for "omit_sensitive" or something like that? I understand why you made the default behavior the safe thing here, and from a security perspective that seems right, but having a default value of type set that is not empty-set for a function where you had to override the definition is just counterintuitive to me... Not blocking, though.

Reply:
Yeah, you're right. It would also be pretty easy to accidentally set your own one field to omit without keeping the sensitive fields' omission. But I also have too much time pressure right now so I'll just merge it as it is. :(